app/pulp/app/serializers/user.py

```diff
@@ -1,19 +1,60 @@
+from gettext import gettext as _
+
+from django.core.exceptions import ObjectDoesNotExist
+from django.contrib.auth.hashers import make_password
+from django.utils import six
 from rest_framework import serializers
 
 from pulp.app import models
 from pulp.app.serializers import ModelSerializer
 
 
+class PasswordSerializer(serializers.CharField):
+    """
+    Serializer for the password field of User object.
+    """
+    def to_internal_value(self, data):
+        # We're lenient with allowing basic numerics to be coerced into strings,
+        # but other types should fail. Eg. unclear if booleans should represent as `true` or `True`,
+        # and composites such as lists are likely user error.
+        if isinstance(data, bool) or not isinstance(data, six.string_types + six.integer_types + (float,)):
+            self.fail('invalid')
+        value = six.text_type(make_password(data))
+        return value
+
+
 class UserSerializer(ModelSerializer):
+    _href = serializers.HyperlinkedIdentityField(view_name='users-detail',
+                                                 lookup_field='username')
+
     username = serializers.CharField(
-        help_text="Required. 150 characters or fewer. Letters, digits and @/./+/-/_ only."
+        help_text="Required. 150 characters or fewer. Letters, digits and @/./+/-/_ only.",
+        read_only=True
     )
 
     is_superuser = serializers.BooleanField(
         help_text="Designates that this user has all permissions without explicitly assigning "
-                  "them."
+                  "them.",
+        required=False
     )
 
+    password = PasswordSerializer(write_only=True, help_text="Password")
+
+    def validate_username(self, value):
+        """
+        Check that the username does not exist in the database.
+        :param value: Username being validated
+        :type value: str
+        :return: value
+        :rtype: str
+        :raises: :class:`rest_framework.serializers.ValidationError`
+        """
+        try:
+            models.PulpUser.objects.get(username=value)
```

Comment:
Existence checks should use `.exists`, e.g. `models.PulpUser.objects.filter(username=value).exists()`

If the username Model field is not indexed, we'll want it to be indexed.

https://docs.djangoproject.com/en/1.8/ref/models/querysets/#django.db.models.query.QuerySet.exists